src/OpenSpaceToolkit/Astrodynamics/Trajectory/Sequence.cpp

```diff
@@ -243,6 +243,15 @@ Sequence::Solution Sequence::solve(
     {
         for (const Segment& segment : segments_)
         {
+            if (segment.accessEventCondition()->getTarget().type_ == EventCondition::Target::Type::RelativeSegmentStart)
+            {
+                segment.accessEventCondition()->updateTarget(initialState);
+            }
+            else if (segment.accessEventCondition()->getTarget().type_ == EventCondition::Target::Type::RelativeSequenceStart)
+            {
+                segment.accessEventCondition()->updateTarget(aState);
```

Comment:
Since this seems like a pretty core part of the flow, can you add a test that covers it 🙏